src/serializer.ts

```diff
@@ -105,9 +105,6 @@ export class Serializer {
             return new clazz();
         }
         let resultConstructor: new() => any = this.getClass(clazz, obj, discriminatorField);
```

Comment:
You can avoid type hinting when it can be inferred from the `this.getClass()` return type.